cmd/juno/juno.go

```diff
@@ -215,3 +238,19 @@ func NewCmd(config *node.Config, run func(*cobra.Command, []string) error) *cobr
 
 	return junoCmd
 }
+
+func defaultDataDir() (string, error) {
+	const junoDirName = "juno"
+
+	if userDataDir := os.Getenv("XDG_DATA_HOME"); userDataDir != "" {
+		return filepath.Join(userDataDir, junoDirName), nil
+	}
+
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	} else if userHomeDir == "" {
+		return "", errors.New("home directory not found")
+	}
+	return filepath.Join(userHomeDir, ".local", "share", junoDirName), nil
```

Comment:
would it make our lives easier if we used `cwd/juno` as the default datadir?

Reply:
I'd love to do that.

Main downside is that restarting Juno in a different working directory with the same params will not use the existing db. We follow the [XDG base directory](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#Environment_variables) spec right now, so `XDG_DATA_HOME` should exist. Wdyt?

Reply:
> Main downside is that restarting Juno in a different working directory with the same params will not use the existing db

I wouldnt mind that. Using cwd sounds reasonable enough to me. cc: @IronGauntlets

Reply:
Using `cwd` sounds good to me. Whenever I run juno I specify the `cwd` as `datadir`.